### ISSUE TITLE
participantLeft returns other participant id when terminate terminate our session

### ISSUE THREAD

**Issue by mukesh4718** [jitsi/jitsi-meet#3720]
I am initiating a call from one user and other user get joined from other browser. When Initiator leave the conference the **participantLeft** is triggered and getting the other user id in arg.id field. As per doc [](https://github.com/jitsi/jitsi-meet/blob/master/doc/api.md) the **participantLeft** event triggered when any participant is left and in arg field the left participant id is returned, but in my case the method returns wrong participant id which is other user id.

**Call Inititor Code:**
`joinSession() {
    this.domain = 'meet.jit.si';
    this.roomName = `${this.tutor['fullName']}_${this.student['fullName']}_${new Date().getTime()}`;
    this.options = {
      roomName: this.roomName,
      width: 800,
      height: 500,
      parentNode: document.querySelector('#jitsiVideo'),
      configOverwrite: {},
      interfaceConfigOverwrite: {
        // filmStripOnly: true,
        TOOLBAR_BUTTONS: [
          'microphone',
          'camera',
          'closedcaptions',
          'desktop',
          'fullscreen',
          'hangup',
          'profile',
          'chat',
          'recording'
        ]
      }
    };

    this.api = new JitsiMeetExternalAPI(this.domain, this.options);
    this.api.executeCommand('displayName', this.tutor['fullName']);

    this.api.addEventListeners({
      readyToClose: this.unload,
      participantLeft: this.handleParticipantLeft,
      participantJoined: this.handleParticipantJoined,
      videoConferenceJoined: this.handleVideoConferenceJoined,
      videoConferenceLeft: this.handleVideoConferenceLeft
    });

    // auto close after 1 min if no participants are joined
    setTimeout(() => {
      if (this.api.getNumberOfParticipants() === 1) {
        this.unload();
        alert(
          'Student did not join the session, hence its terminated automatically. You can re-initiate call again.'
        );
      }
    }, 60000 * 5);
    this.conferenceJoined = true;
    this.jitsiCall = 'start';
  }
  handleParticipantLeft = arg => {
    console.log('tutor participant left is called: ', arg);
    this.jitsiService
      .getJitsiDetailByParticipantId(arg.id)
      .subscribe(async roomDetail => {
        if (!roomDetail) {
          console.log(`No room is joined by the participant id: ${arg.id}`);
          return false;
        } else {
          const participantDetail = roomDetail['participants'].filter(
            el => el.jitsiParticipantId === arg.id
          )[0];

          if (participantDetail) {
            switch (participantDetail.type) {
              case 'manager':
                console.log('Left participant is manager');
                break;
              case 'supervisor':
                console.log('Left participant is supervisor');
                break;
              case 'student':
                console.log('Left participant is student');
                // const obj = {
                //   participantId: arg.id,
                //   roomName: this.roomName,
                //   tutorId: this.tutor['_id'],
                //   type: participantDetail.type
                // };
                // await this.jitsiService.leaveJitsiRoom(obj);
                // // terminate tutor session
                // this.api.dispose();
                swal('Student left the session.');
                this.unload();
                // this.conferenceJoined = false;
                break;
              case 'tutor':
                console.log('Left participant is tutor');
                break;
              default:
                console.log('Left participant is not a valid type');
                break;
            }
          }
        }
      });
  };
  handleParticipantJoined = arg => {
    console.log('participant joined: ', arg, this.api);
  };
  handleVideoConferenceJoined = arg => {
    const obj = {
      participantId: arg.id,
      roomName: this.roomName,
      tutorId: this.tutor['_id'],
      tutorJoined: 'yes',
      studentId: this.student['_id'],
      studentJoined: 'no'
    };
    // save new room
    this.jitsiService.addJitsiRoom(obj);
  };
  handleVideoConferenceLeft = arg => {};

  unload = () => {
    this.jitsiCall = 'end';
    if (this.api instanceof JitsiMeetExternalAPI) {
      this.api.dispose();
      const obj = {
        participantId: this.api._myUserID,
        roomName: this.roomName,
        tutorId: this.tutor['_id'],
        type: 'tutor'
      };
      this.jitsiService.leaveJitsiRoom(obj);
    }
    $('#jitsiVideo').html('');
    this.conferenceJoined = false;
  };`

**Call Receiver Code:**

`joinSession(sessionData: any) {
    if (sessionData) {
      // initialize variables
      this.roomName = sessionData['roomName'];
      this.loggedUser = sessionData['loggedUser'];
      this.otherUser = sessionData['otherUser'];
      this.conferenceJoined = false;
      this.domain = 'meet.jit.si';
      this.options = {
        roomName: this.roomName,
        width: 800,
        height: 500,
        parentNode: document.querySelector('#jitsiVideo'),
        configOverwrite: {},
        interfaceConfigOverwrite: {
          // filmStripOnly: true,
          TOOLBAR_BUTTONS: [
            'microphone', 'camera', 'closedcaptions', 'desktop', 'fullscreen',
            'hangup', 'profile', 'chat', 'recording'
          ],
        }
      };

      try {
        this.api = new JitsiMeetExternalAPI(this.domain, this.options);
        this.api.executeCommand('displayName', this.loggedUser['fullName']);

        this.api.addEventListeners({
          readyToClose: this.unload,
          // participantLeft: this.handleParticipantLeft,
          participantJoined: this.handleParticipantJoined,
          videoConferenceJoined: this.handleVideoConferenceJoined,
          videoConferenceLeft: this.handleVideoConferenceLeft
        });

        this.conferenceJoined = true;
      } catch (error) {
        console.log(error.message);
      }
    }
  }
  unload = () => {
    if (this.api instanceof JitsiMeetExternalAPI) {
      this.api.dispose();
    }
    $('#jitsiVideo').html('');
    this.conferenceJoined = false;
    this.hasActiveRoom = false;
  }
  handleParticipantLeft = (arg) => {
    console.log('participant left: ', arg);
    this.getJitsiDetailByParticipantId(arg.id)
      .subscribe(async roomDetail => {
        if (!roomDetail) {
          console.log(`No room is joined by the participant id: ${arg.id}`);
          return false;
        } else {
          const participantDetail = roomDetail['participants'].filter(el => el.jitsiParticipantId === arg.id)[0];

          if (participantDetail) {
            switch (participantDetail.type) {
              case 'manager':
                console.log('Left participant is manager');
                break;
              case 'supervisor':
                console.log('Left participant is supervisor');
                break;
              case 'student':
                console.log('Left participant is student');
                break;
              case 'tutor':
                console.log('Left participant is tutor');
                this.api.dispose();
                this.conferenceJoined = false;
                this.hasActiveRoom = false;
                alert('Tutor left the session.');
                break;
              default:
                console.log('Left participant is not a valid type');
                break;
            }
          }
        }
      });
  }
  handleParticipantJoined = (arg) => {
    console.log('participant joined: ', arg);
  }
  handleVideoConferenceJoined = (arg) => {
    console.log(`video conference joined: `, arg, 'otherUser: ', this.otherUser, 'loggedUser: ', this.loggedUser);
    const obj = {
      participantId: arg.id,
      roomName: arg.roomName,
      tutorId: this.otherUser[`_id`],
      studentId: this.loggedUser[`_id`],
      studentJoined: 'yes',
      joineeType: 'student',
    }
    // save new room
    this.updateJitsiRoomForStudent(obj);
  }
  handleVideoConferenceLeft = (arg) => {
    console.log(`video conference left: `, arg);
  }`

**Comment by saghul** [comment]
I don't think this is incorrect behavior, but incomplete documentation. When you leave the conference all participants "leave" too (from the user's perpective). So you should see events for everyone in the conference.

**Comment by krishnap7891** [comment]
how can I know that all participate is leaved and I am alone in meeting in jitsmeet sdk ??
